test/conftest.py

```diff
@@ -262,6 +266,43 @@ def attribute_single_value_pow2(classname, devicename, attr, max_pow, tol, repea
         assert bi.dev_interface(val, attr, tol) <= tol
 
 
+def attribute_multipe_values(classname, devicename, attr, values, tol, repeats=1):
+    bi = BoardInterface(classname, devicename)
+    for _ in range(repeats):
+        for val in values:
+            if isinstance(val, str):
+                assert bi.dev_interface(val, attr, 0)
+            else:
+                assert bi.dev_interface(val, attr, tol) <= tol
+
+
+def attribute_multipe_values_with_depends(classname, devicename, attr, depends, values, tol, repeats=1):
+    bi = BoardInterface(classname, devicename)
+    # Set custom dependencies for the attr being tested
+    for p in depends.keys():
+        if isinstance(depends[p], str):
+            assert bi.dev_interface(depends[p], p, 0)
+        else:
+            assert bi.dev_interface(depends[p], p, tol) <= tol
+    for _ in range(repeats):
+        for val in values:
+            if isinstance(val, str):
+                assert bi.dev_interface(val, attr, 0)
+            else:
+                assert bi.dev_interface(val, attr, tol) <= tol
+
+
+def attribute_profile_write(classname, devicename, attr, file):
```

Comment:
Can we make this a little more generic? Maybe call it attribute_write_only_str or something.